src/yunohost/user.py

```diff
@@ -398,20 +398,44 @@ def user_info(auth, username):
         result_dict['mail-forward'] = user['maildrop'][1:]
 
     if 'mailuserquota' in user:
-        if user['mailuserquota'][0] != '0': 
-           cmd = 'doveadm -f flow quota get -u %s' % user['uid'][0]
-           userquota = subprocess.check_output(cmd,stderr=subprocess.STDOUT,
-                                             shell=True)
-           quotavalue = re.findall(r'\d+', userquota)
-           result = '%s (%s%s)' % ( _convertSize(eval(quotavalue[0])), 
-                                             quotavalue[2], '%')
-           result_dict['mailbox-quota'] = {
-               'limit' : user['mailuserquota'][0],
-               'use' : result
-           }
-        else:
-           result_dict['mailbox-quota'] = m18n.n('unlimit')
- 
+        userquota = user['mailuserquota'][0]
+
+        if isinstance(userquota, int):
+            userquota = str(userquota)
+
+        # Test if userquota is '0' or '0M' ( quota pattern is ^(\d+[bkMGT])|0$ )
+        is_limited = not re.match('0[bkMGT]?', userquota)
+        storage_use = '?'
+
+        cmd = 'doveadm -f flow quota get -u %s' % user['uid'][0]
+
+        try:
+            cmd_result = subprocess.check_output(cmd, stderr=subprocess.STDOUT,
+                                                 shell=True)
+            # Exemple of return value for cmd:
+            # """Quota name=User quota Type=STORAGE Value=0 Limit=- %=0
+            # Quota name=User quota Type=MESSAGE Value=0 Limit=- %=0"""
+            has_value = re.search(r'Value=(\d+)', cmd_result)
+
+            if has_value:
+                storage_use = int(has_value.group(1))
+                storage_use = _convertSize(storage_use)
+
+                if is_limited:
+                    has_percent = re.search(r'%=(\d+)', cmd_result)
+
+                    if has_percent:
+                        percentage = int(has_percent.group(1))
+                        storage_use += ' (%s%%)' % percentage
+
+        except subprocess.CalledProcessError:
```

Comment:
Quoting python documentation `If the return code was non-zero it raises a CalledProcessError.`. This mean that **any** non zero value will raise this and our code consider that this always mean that dovecot is down which could not be the case, this will create a very annoying to debug situation :/